src/util/http.ts

```diff
@@ -0,0 +1,56 @@
+import { AxiosRequestConfig } from 'axios';
+
+/**
+ * Gets the base HTTP request config to use for all REST API requests.
+ * @param {string | undefined} session - Session to use for authentication. Only used for Cloud endpoints.
+ * @returns {AxiosRequestConfig} - Request configuration compatible with Axios.
+ */
+function getBaseRequestConfig(session: string | undefined): AxiosRequestConfig {
+  const requestConfig: AxiosRequestConfig = {};
+  if (session) {
+    requestConfig['headers'] = {
+      Cookie: `session=${session}`,
+    };
+  }
+  return requestConfig;
+}
+
+/**
+ * Gets the value of the given cookie key from all response cookies.
+ * @see {@link https://datatracker.ietf.org/doc/html/rfc6265#section-3.1|Cookies RFC} for example of cookie syntax.
+ * @param {Array<string>} allCookies - All response cookies.
+ * @param {string} cookieKey - Name of cookie to retrieve.
+ * @returns {string | undefined} - Value of given cookie, or undefined if not present.
+ */
+function getCookieValue(
+  allCookies: Array<string>,
+  cookieKey: string
+): string | undefined {
+  for (let i = 0; i < allCookies.length; i++) {
```

Comment:
what do you think about reduce the complexity in this function? I belive could be a good option reduce this one with arrays methods as map